### PR TITLE
(Browser) Support for File Uri Upload

### PR DESCRIPTION
**Pull request by dplewis**
`http` and `https` modules don't exist in browser

### CODE REVIEW

File: src/ParseFile.js [hunk 1]
```diff
@@ -317,6 +317,9 @@ const DefaultController = {
     if (XHR) {
       return this.downloadAjax(uri);
     }
+    if (process.env.PARSE_BUILD !== 'node') {
```
Comment (acinader) [inline src/ParseFile.js]: maybe i don't understand what process.env.PARSE_BUILD is, but the way I'm reading this is:

if we're not in node, complain that we don't have XMLHttpRequest, but XMLHttpRequest is in the browser, not node.  set me straight here :).

Comment (dplewis) [reply src/ParseFile.js]: For browsers that don't support ajax. 

https://www.tutorialspoint.com/ajax/ajax_browser_support.htm

Comment (dplewis) [reply src/ParseFile.js]: PARSE_BUILD is set when building the sdk. 

https://github.com/parse-community/Parse-SDK-JS/blob/master/build_releases.sh

File: src/ParseFile.js [hunk 2]
```diff
@@ -311,10 +314,16 @@ const DefaultController = {
   },
 
   download: function(uri) {
+    if (XHR) {
```
Comment (acinader) [inline src/ParseFile.js]: I think I get it.  Fine as is.

re-writting below how I would formultate.  You think this is functionally the same?

no need to change, just curious.
```
if(process.env.PARSE_BUILD === 'browser') {
  if(!XHR) {
     return Promise.reject('Cannot make a request: No definition of XMLHttpRequest was found.');
  }

  return return this.downloadAjax(uri);
}
// We're in node
return new Promise((resolve, reject) => {
    const client = uri.indexOf('https') === 0 
        ? require('https')
        : require('http');
...
```

Comment (dplewis) [reply src/ParseFile.js]: Functionality will be the same

Comment (dplewis) [reply src/ParseFile.js]: Actually xmlhttprequest exists in react-native

**Pull request merged**